### ISSUE TITLE
need intuitive explanation of K, v, a , h

### ISSUE THREAD

**Issue by dragen1860** [whyjay/memoryGAN#2]
Hi, dear author:
  Thanks for your work on memory topics.
  When i try to digest your paper, however, I found it's very difficult to understand the role of four tuples`(K, v, a, h)`. I dnt know what's the intuitive function of these variables. Although it does works well according to your exp., some intuitive explanation of these variable will helps much.

**Comment by whyjay** [comment]
Hi, 
Here's brief explanation about each components.
I hope this helps you understanding MemoryGAN.

1) K : key
As our memory is a vMF mixture distribution, each memory slot is a mixture component.
Thus, K is mean direction of a vMF distribution.
Note that Norm(K) = 1 since vMF is defined on unit hypersphere.
(or equivalently, in terms of memory network, K is the key vector of a memory slot)

2) v : value
v is a binary variable (0 or 1) which indicates whether the vMF component is about real (=1) samples or not (=0).

3) a : age
a is an scalar number. Whenever memory is updated, all not-modified memory slots increase their age by one. We initialize maximum-age memory slot when we apply Least Recently Used (LRU) memory update. 

4) h : histogram
h is an scalar number. You may want to interpret it as 'effective number of samples within the vMF component (or the memory slot)', or to be precise, it is accumulated posterior.

**Comment by dragen1860** [comment]
Thx for your detailed explanation.
I wonder will the four tuples (K, v, a, h) be optimized by Optimizer, Since you used lots of `tf.stop_gradients` in implemnetation?

**Comment by whyjay** [comment]
(K, v, a, h) is not trainable tensor. (We don't optimize them through gradient descent)
For example, K is updated only by LRU or incremental EM-like update rule.

**Comment by dragen1860** [comment]
Will u continue to do research on this paper?  or on memory topic generally?

**Comment by whyjay** [comment]
Currently, i'm working on reinforcement learning research.
I might come back to generative models and memory things, 
but i am not working on those topics now.

**Comment by dragen1860** [comment]
I see, Thx for you explanation.
